platform/linux-generic/include/odp_pool_internal.h

```diff
@@ -29,10 +29,15 @@ extern "C" {
 
 #define CACHE_BURST    32
 
-typedef struct pool_cache_t {
+struct pool_cache_s {
 	uint32_t num;
 	uint32_t buf_index[CONFIG_POOL_CACHE_SIZE];
 
+};
+
+typedef union pool_cache_u {
+	struct pool_cache_s s;
+	uint8_t pad[ROUNDUP_CACHE_LINE(sizeof(struct pool_cache_s))];
 } pool_cache_t ODP_ALIGNED_CACHE;
```

Comment:
With Kevin's latest change (PR #326) this is already an integral multiple of the cache size. Not sure what value we're gaining here. Is this to guard against random changes to `CONFIG_POOL_CACHE_SIZE`? That still could leave a lot of wasted space, which was the idea behind Kevin's PR.

Reply:
Kevin's PR is to adjust the default value so that by default we are using 1 cache line. However, CONFIG_POOL_CACHE_SIZE is configurable. If the user is not aware of the internal details of the pool implementation, parts of 2 elements of the pool cache array will be in the same cache line. This change will guard against that. If the CONFIG_POOL_CACHE_SIZE is configured correctly, this will not add any additional space.

Reply:
My point is that we want two things: Cache alignment and no wasted space. Kevin's PR addressed this issue. It seems all we need is document the implications for those who wish to set `CONFIG_POOL_CACHE_SIZE` since otherwise we'll just be inserting a lot of padding that once again will waste space.